api/rds.proto

```diff
@@ -227,14 +225,16 @@ message RouteAction {
     // 2. Generated. Envoy generates and sets a cookie with an expiration (TTL)
     //    on the first request from the client in its response to the client,
     //    based on the endpoint the request gets sent to. The client then
-    //    presents this on the next and all subsequent requests, the hash of
+    //    presents this on the next and all subsequent requests. The hash of
     //    this is sufficient to ensure these requests get sent to the same
-    //    endpoint.
+    //    endpoint. The cookie is generated by hashing the source and
+    //    destination ports and addresses so that multiple independent HTTP2
+    //    streams will independently receive the same cookie, even if they
```

Comment:
streams on the same connection. If you had a "smart" gRPC client that initiated multiple connections (and somehow had a cookie store, which gRPC clients don't tend to), there might still be a race.